app/scripts/metamask-controller.js

```diff
@@ -1826,12 +1827,18 @@ export default class MetamaskController extends EventEmitter {
    * @param {Function} cb - The callback function called with the signature.
    */
   newUnsignedMessage(msgParams, req) {
-    const promise = this.messageManager.addUnapprovedMessageAsync(
-      msgParams,
-      req,
-    );
-    this.sendUpdate();
-    this.opts.showUserConfirmation();
+    const data = normalizeMsgData(msgParams.data);
```

Comment:
Just as with the other comment, I think having two styles of throwing errors in this function is confusing and should be avoided. Making this `async` would ensure errors are thrown in a single way.

This was an issue that I had known about for a while, and haven't gotten around to fixing yet.